webcompat/static/js/lib/bugform.js

```diff
@@ -75,13 +75,18 @@ function BugForm() {
       _.bind(this.checkDescriptionValidity, this)
     );
     this.problemType.on("change", _.bind(this.checkProblemTypeValidity, this));
+    this.stepsToReproduceField.on(
+      "focus",
+      _.bind(this.clearStepsToReproducePlaceholder, this)
+    );
     this.uploadField.on("change", _.bind(this.checkImageTypeValidity, this));
     this.osField
       .add(this.browserField)
       .on("blur input", _.bind(this.checkOptionalNonEmpty, this));
     this.submitButtons.on("click", _.bind(this.storeClickedButton, this));
     this.submitButtons.on("click", _.bind(this.loadingIndicator.show, this));
     this.form.on("submit", _.bind(this.maybeUploadImage, this));
+    this.form.on("submit", _.bind(this.clearStepsToReproducePlaceholder, this));
```

Comment:
No, not quite. I would re-name `maybeUploadImage` to something less image specific (maybe just `onFormSubmit`) and call `clearStepsToReproducePlaceholder` inside that method.